cartography/intel/aws/identitycenter.py

```diff
@@ -0,0 +1,371 @@
+                    permission_sets.append(details.get('PermissionSet', {}))
+                except client.exceptions.ClientError as e:
+                    logger.warning(f"Failed to get details for permission set {arn}: {e}")
+    except client.exceptions.ClientError as e:
+        logger.warning(f"Failed to get permission sets for instance {instance_arn} in region {region}: {e}")
+        return []
+
+    return permission_sets
+
+
+@timeit
+def get_permission_set_roles(
+    boto3_session: boto3.session.Session,
+    instance_arn: str,
+    permission_set_arn: str,
+    region: str,
+) -> List[Dict]:
+    """
+    Get all accounts associated with a given permission set
+    """
+    client = boto3_session.client('sso-admin', region_name=region)
+    accounts = []
+
+    try:
+        paginator = client.get_paginator('list_accounts_for_provisioned_permission_set')
+        for page in paginator.paginate(InstanceArn=instance_arn, PermissionSetArn=permission_set_arn):
+            accounts.extend(page.get('AccountIds', []))
+    except client.exceptions.ClientError as e:
+        logger.warning(f"Failed to get roles for permission set {permission_set_arn}: {e}")
+        return []
+
+    return accounts
+
+
+@timeit
+def load_permission_sets(
+    neo4j_session: neo4j.Session,
+    permission_sets: List[Dict],
+    instance_arn: str,
+    region: str,
+    aws_update_tag: int,
+) -> None:
+    """
+    Load Identity Center permission sets into the graph
+    """
+    logger.info(f"Loading {len(permission_sets)} permission sets for instance {instance_arn} in region {region}")
+
+    load(
+        neo4j_session,
+        PermissionSetSchema(),
+        permission_sets,
+        lastupdated=aws_update_tag,
+        InstanceArn=instance_arn,
+        Region=region,
+    )
+
+
+@timeit
+def get_permission_set_role_assignments(
+    boto3_session: boto3.session.Session,
+    instance_arn: str,
+    permission_sets: List[Dict],
+    region: str,
+) -> List[Dict]:
+    """
+    Get role assignments for Identity Center permission sets
+    """
+    role_assignments = []
+    for ps in permission_sets:
+        permission_set_arn = str(ps.get('PermissionSetArn'))
+        accounts = get_permission_set_roles(boto3_session, instance_arn, permission_set_arn, region)
+        for account_id in accounts:
+            arn = f"arn:aws:iam::{account_id}:role/aws-reserved/sso.amazonaws.com/AWSReservedSSO_{ps.get('Name')}"
+            role_assignments.append({
+                'PermissionSetArn': ps.get('PermissionSetArn'),
+                'RoleArn': arn,
+            })
+    return role_assignments
+
+
+@timeit
+def load_permission_set_role_assignments(
+    neo4j_session: neo4j.Session,
+    role_assignments: List[Dict],
+    aws_update_tag: int,
+) -> None:
+    """
+    Load Identity Center permission set role assignments into the graph
+    """
+    logger.info(f"Loading {len(role_assignments)} permission set role assignments")
+
+    neo4j_session.run(
```

Comment:
Same thing here about using the data model and a transform step.

Reply:
Understood. I took a stab at trying to allow fuzzy matching in the data model: https://github.com/cartography-cncf/cartography/pull/1383 - let me know if this helps